File: contracts/src/main/java/my/restate/e2e/services/KillTest.java
// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
//
// This file is part of the Restate e2e tests,
// which are released under the MIT license.
//
// You can find a copy of the license in file LICENSE in the root
// directory of this repository or package, or at
// https://github.com/restatedev/e2e/blob/main/LICENSE

package my.restate.e2e.services;

import dev.restate.sdk.Context;
import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;
import dev.restate.sdk.annotation.VirtualObject;

public interface KillTest {

  @Service
  interface Runner {
    @Handler
    void startCallTree(Context context);
  }

  @VirtualObject
  interface Singleton {
    @Handler
    void recursiveCall(ObjectContext context);

    @Handler
    void isUnlocked(ObjectContext context);
  }
}

File: sdk-tests/src/main/kotlin/dev/restate/sdktesting/contracts/KillTest.kt
// Copyright (c) 2023 - Restate Software, Inc., Restate GmbH
//
// This file is part of the Restate SDK Test suite tool,
// which is released under the MIT license.
//
// You can find a copy of the license in file LICENSE in the root
// directory of this repository or package, or at
// https://github.com/restatedev/sdk-test-suite/blob/main/LICENSE
package dev.restate.sdktesting.contracts

import dev.restate.sdk.annotation.*

interface KillTest {
  @VirtualObject
  @Name("KillTestRunner")
  interface Runner {
    @Handler suspend fun startCallTree()
  }

  @VirtualObject
  @Name("KillTestSingleton")
  interface Singleton {
    @Handler suspend fun recursiveCall()

    @Handler suspend fun isUnlocked()
  }
}
